components/director/internal/domain/api/repository_test.go

```diff
@@ -0,0 +1,327 @@
+		assert.Equal(t, firstApiDefID, modelAPIDef.Data[0].ID)
+		assert.Equal(t, secondApiDefID, modelAPIDef.Data[1].ID)
+		assert.Equal(t, "", modelAPIDef.PageInfo.StartCursor)
+		assert.Equal(t, totalCount, modelAPIDef.TotalCount)
+		convMock.AssertExpectations(t)
+		sqlMock.AssertExpectations(t)
+	})
+
+	t.Run("returns error when conversion from entity to model failed", func(t *testing.T) {
+		testErr := errors.New("test error")
+		rows := sqlmock.NewRows(fixApiDefinitionColumns()).
+			AddRow(fixApiDefinitionRow(firstApiDefID, "placeholder")...)
+
+		sqlMock.ExpectQuery(selectQuery).
+			WithArgs(tenantID).
+			WillReturnRows(rows)
+
+		sqlMock.ExpectQuery(countQuery).
+			WithArgs(tenantID).
+			WillReturnRows(testdb.RowCount(1))
+		ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+
+		convMock := automock.APIDefinitionConverter{}
+		convMock.On("FromEntity", firstApiDefEntity).Return(model.APIDefinition{}, testErr).Once()
+		pgRepository := api.NewPostgresRepository(&convMock)
+		//WHEN
+		_, err := pgRepository.ListByApplicationID(ctx, tenantID, appID, inputPageSize, inputCursor)
+		//THEN
+		require.Error(t, err)
+		require.Contains(t, err.Error(), testErr.Error())
+		convMock.AssertExpectations(t)
+		sqlMock.AssertExpectations(t)
+	})
+}
+
+func TestPgRepository_Create(t *testing.T) {
+	//GIVEN
+	id := fixApiDefID()
+	tenantID := fixTenantID()
+	apiDefModel := fixFullModelAPIDefinition("placeholder")
+	apiDefEntity := fixFullEntityAPIDefinition(id, "placeholder")
+	insertQuery := `^INSERT INTO "public"."api_definitions" \(.+\) VALUES \(.+\)$`
+	sqlxDB, sqlMock := testdb.MockDatabase(t)
+
+	t.Run("success", func(t *testing.T) {
+		defAuth, err := json.Marshal(apiDefModel.DefaultAuth)
+		require.NoError(t, err)
+
+		sqlMock.ExpectExec(insertQuery).
+			WithArgs(fixApiCreateArgs(id, string(defAuth), apiDefModel)...).
+			WillReturnResult(sqlmock.NewResult(-1, 1))
+
+		ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+		convMock := automock.APIDefinitionConverter{}
+		convMock.On("ToEntity", *apiDefModel).Return(*apiDefEntity, nil).Once()
+		pgRepository := api.NewPostgresRepository(&convMock)
+		//WHEN
+		err = pgRepository.Create(ctx, tenantID, apiDefModel)
+		//THEN
+		require.NoError(t, err)
+		sqlMock.AssertExpectations(t)
+		convMock.AssertExpectations(t)
+	})
+
+	t.Run("returns error when conversion from model to entity failed", func(t *testing.T) {
+		ctx := context.TODO()
+		convMock := automock.APIDefinitionConverter{}
+		convMock.On("ToEntity", *apiDefModel).Return(api.APIDefinition{}, errors.New("test error"))
+		pgRepository := api.NewPostgresRepository(&convMock)
+		// WHEN
+		err := pgRepository.Create(ctx, tenantID, apiDefModel)
+		// THEN
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "test error")
+	})
+}
+
+func TestPgRepository_CreateMany(t *testing.T) {
+	tenantID := fixTenantID()
+	insertQuery := `^INSERT INTO "public"."api_definitions" \(.+\) VALUES \(.+\)$`
```

Comment:
Sometimes parentheses are escaped and sometimes not, can you check if it's required and unify it?